client/src/components/SearchCombo.js

```diff
@@ -1,68 +1,79 @@
 // @flow
 import React from 'react';
+import { connect } from 'react-redux';
 import Dropdown from 'react-bootstrap/Dropdown';
 import FormControl from 'react-bootstrap/FormControl';
 
 import { BackendClient } from '../requests';
+import { updateSearchValue, doFetchPlaces } from '../actions';
 
 let connection = new BackendClient();
 
 type Props = {|
   style?: string,
-  className?: string
+  className?: string,
+  updateSearchValue: (string) => void,
+  doFetchPlaces: () => void
 |};
 
 type State = {|
   value: string,
   tags: ?[]
 |};
 
-export class SearchCombo extends React.Component<Props, State> {
+class SearchCombo extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = { value: '', tags: null };
-    this.getPlaces();
+    // this.getPlaces();
+    this.props.updateSearchValue('');
   }
   timer: TimeoutID;
 
   handleChange = (e: SyntheticInputEvent<*>) => {
+    // Called whenever the value in the searchbox changes
     this.setState(
       { value: e.target.value },
       () => {
         if (this.timer) {
           clearTimeout(this.timer);
         }
-        if (this.state.value.length !== 1) {
+        if (this.state.value.length === 0) {
+          // Searchbox is empty, so clear the tags and reload the list of places
+          this.setState(
+            { tags: null },
+            () => {
+              this.refreshPlaces();
+            }
+          );
+        }
+        // Searchbox has at least one character, so fetch matching tags
+        if (this.state.value.length > 1) {
           this.timer = setTimeout(() => this.getTags(), 200);
```

Comment:
Lodash's [debounce](https://lodash.com/docs/4.17.15#debounce) might be useful here :)